contracts/oracles/VelodromeV2Oracle.sol

```diff
@@ -0,0 +1,88 @@
+// SPDX-License-Identifier: MIT
+// solhint-disable one-contract-per-file
+
+pragma solidity 0.8.23;
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "../interfaces/IOracle.sol";
+import "@openzeppelin/contracts/utils/math/Math.sol";
+import "../libraries/OraclePrices.sol";
+
+interface IVelodramoV2Router {
+    function getReserves(address tokenA, address tokenB, bool stable, address _factory) external view returns (uint256 reserveA, uint256 reserveB);
+}
+
+interface IVelodramoV2Registry {
+    function poolFactories() external view returns (address[] memory);
+}
+
+
+contract VelodromeV2Oracle is IOracle {
+    using OraclePrices for OraclePrices.Data;
+    using Math for uint256;
+
+    IERC20 private constant _NONE = IERC20(0xFFfFfFffFFfffFFfFFfFFFFFffFFFffffFfFFFfF);
+
+    IVelodramoV2Router public immutable ROUTER;
+    IVelodramoV2Registry public immutable REGISTRY;
+
+    constructor(IVelodramoV2Router _router, IVelodramoV2Registry _registry) {
+        ROUTER = _router;
+        REGISTRY = _registry;
+    }
+
+    function getRate(IERC20 srcToken, IERC20 dstToken, IERC20 connector, uint256 thresholdFilter) external view override returns (uint256 rate, uint256 weight) {
```

Comment:
Did you intentionally choose not to write natspec docs for some reason?

Reply:
That's a good observation. It seems like a useful idea to add NatSpec comments, at least in the interface, to enhance documentation and understanding